rift/node.py

```diff
@@ -2469,87 +2517,116 @@ def spf_run_direction(self, spf_direction):
+        prefix_ties = self.ties_of_type(direction=self.spf_use_tie_direction(node_sysid, spf_direction),
+                                        system_id=node_sysid,
+                                        prefix_type=common.ttypes.TIETypeType.PrefixTIEType
+                                        )
 
-    def spf_add_prefixes(self, node_sysid, node_cost, candidates, spf_direction):
-        prefix_ties = self.ties_of_type(
-            direction=self.spf_use_tie_direction(node_sysid, spf_direction),
-            system_id=node_sysid,
-            prefix_type=common.ttypes.TIETypeType.PrefixTIEType)
         for prefix_tie in prefix_ties:
-            self.spf_add_prefixes_common(
-                node_sysid, node_cost, candidates, spf_direction,
-                prefix_tie.element.prefixes.prefixes, False)
-
-    def spf_add_pos_disagg_prefixes(self, node_sysid, node_cost, candidates, spf_direction):
-        prefix_ties = self.ties_of_type(
-            direction=self.spf_use_tie_direction(node_sysid, spf_direction),
-            system_id=node_sysid,
-            prefix_type=common.ttypes.TIETypeType.PositiveDisaggregationPrefixTIEType)
+            self.spf_add_prefixes_common(node_sysid, node_cost, candidates, spf_direction, special_for_neg_disagg,
+                                         prefix_tie.element.prefixes.prefixes,
+                                         is_pos_disagg=False
+                                         )
+
+    def spf_add_pos_disagg_prefixes(self, node_sysid, node_cost, candidates, spf_direction, special_for_neg_disagg):
+        prefix_ties = self.ties_of_type(direction=self.spf_use_tie_direction(node_sysid, spf_direction),
+                                        system_id=node_sysid,
+                                        prefix_type=common.ttypes.TIETypeType.PositiveDisaggregationPrefixTIEType
+                                        )
+
         for prefix_tie in prefix_ties:
-            self.spf_add_prefixes_common(
-                node_sysid, node_cost, candidates, spf_direction,
-                prefix_tie.element.positive_disaggregation_prefixes.prefixes, True)
-
-    def spf_add_prefixes_common(self, node_sysid, node_cost, candidates, spf_direction, prefixes,
-                                is_pos_disagg):
-        if prefixes:
-            for prefix, attributes in prefixes.items():
-                tags = attributes.tags
-                cost = node_cost + attributes.metric
-                dest = spf_dest.make_prefix_dest(prefix, tags, cost, is_pos_disagg)
-                self.spf_consider_candidate_dest(dest, None, node_sysid, candidates, spf_direction)
+            self.spf_add_prefixes_common(node_sysid, node_cost, candidates, spf_direction, special_for_neg_disagg,
+                                         prefix_tie.element.positive_disaggregation_prefixes.prefixes,
+                                         is_pos_disagg=True
+                                         )
```

Comment:
Don't put closing bracket on separate line

PS: I am not going to repeat this comment any more; please apply to remaining code as well.